src/gpu/intel/ocl/reusable_softmax.cl

```diff
@@ -72,3 +74,140 @@ reusable_softmax_fwd_generic(__global SRC_DATA_T *src, __global DST_DATA_T *dst,
         dst[c - begin] = TO_DST(unscaled * scale);
     }
 }
+
+#endif
+
+#ifdef USE_VECTORIZED_KERNEL
+
+__attribute__((reqd_work_group_size(SUBGROUP_SIZE, 1, 1)))
+__attribute__((intel_reqd_sub_group_size(SUBGROUP_SIZE))) __kernel void
+reusable_softmax_fwd_generic(__global DATA_T *src, __global DST_DATA_T *dst,
+        __global float *src_scale, __global float *dst_scale,
+        dim_t SOFTMAX_AXIS_SIZE, dim_t softmax_axis_stride,
+        dim_t softmax_axis_chunk_size, dispatch_gws_rt_params_t gws_params) {
+
+    float scale = 1.0f;
+
+    const off_t linear_thread_id = get_global_id(0);
+    const off_t data_off
+            = (linear_thread_id / SUBGROUP_SIZE) * SOFTMAX_AXIS_SIZE;
+    global DATA_T *src_backup = src;
+    src += data_off;
+
+    VECT_FLOAT_T dk;
+    float max_ = -INFINITY;
+    float denom_ = 0.0f;
+    const bool has_tail = SOFTMAX_AXIS_SIZE % (SUBGROUP_SIZE * 8);
+    int last_buf = div_up(SOFTMAX_AXIS_SIZE, (SUBGROUP_SIZE * 8));
+    if (has_tail) last_buf--;
+
+    for (int k = 0; k < last_buf; ++k) {
+        off_t idx = k * SUBGROUP_SIZE;
+        dk = CONVERT_VECT_FLOAT_T(AS_VECT_DATA_T(
+                VECT_BLOCK_READ((const __global BLOCK_DATA_T *)&src[idx * 8])));
+        for (int i = 0; i < 8; ++i) {
```

Comment:
Use a named constant instead of 8, so that we can easily modify this value in the future.